src/connectionimpl.cpp

```diff
@@ -123,7 +123,9 @@ uint64_t ConnectionImpl::parse(const Buffer &buffer)
         try
         {
             // try to recognize the frame
-            ReceivedFrame receivedFrame(ReducedBuffer(buffer, processed), _maxFrame);
```

Comment:
Actually, copying doesn't occur here. I used that syntax to avoid so called *most vexing parse*, but if you have some doubts, you could use this code snippet which completely resembles the original code.
http://pastebin.com/erL8GGXs